src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42RttSpec.groovy

```diff
@@ -1,96 +1,96 @@
-        def statsData = null
-        Wrappers.wait(STATS_LOGGING_TIMEOUT, 1) {
-            statsData = otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+        then: "Check if stats for forward are available"
+        Wrappers.wait(WAIT_OFFSET, 1) {
+            def statsData = otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
                     [flowid   : flow.flowId,
                      direction: "forward"]).dps
             assert statsData && !statsData.empty
         }
 
-        and: "Check if stats for reverse is available"
-        Wrappers.wait(STATS_LOGGING_TIMEOUT, 1) {
-            statsData = otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
+        and: "Check if stats for reverse are available"
+        Wrappers.wait(WAIT_OFFSET, 1) {
+            def statsData = otsdb.query(flowCreateTime, metricPrefix + "flow.rtt",
                     [flowid   : reversedFlow.flowId,
                      direction: "reverse"]).dps
             assert statsData && !statsData.empty
         }
 
-        and: "Cleanup: revert system to original state"
-        revertToOrigin(flow, reversedFlow, flowRttFeatureStartState, flowRttSwitchFeatureStartState, server42Switch)
+        cleanup: "Revert system to original state"
+        revertToOrigin(flow, reversedFlow, flowRttFeatureStartState, initialSwitchRtt)
     }
 
-    def "Create two flow with server42 Rtt feature and check feature togglers"() {
-        given: "Two active neighboring switches one with server42 and one without"
-        def server42switches = topology.getActiveServer42Switches();
-        def server42switchesDpIds = server42switches*.dpId;
-        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find {
+    @Tidy
+    def "Flow rtt stats are available only if both global and switch toggles are 'on' on both endpoints"() {
+        given: "Two active switches with src switch having server42"
+        def server42switches = topology.getActiveServer42Switches()
+        def server42switchesDpIds = server42switches*.dpId
+        def switchPair = topologyHelper.switchPairs.collectMany { [it, it.reversed] }.find {
             it.src.dpId in server42switchesDpIds && !server42switchesDpIds.contains(it.dst.dpId)
         }
         assumeTrue("Was not able to find a switch with a server42 connected", switchPair != null)
         def server42Switch = switchPair.src
+        def statsWaitSeconds = 4
 
-        when: "Set server42FlowRtt toggle to false"
-        def flowRttFeatureStartState = isEnabledFlowRtt();
-        disableFlowRtt()
+        and: "server42FlowRtt toggle is turned off"
+        def flowRttFeatureStartState = changeFlowRttToggle(false)
 
-        and: "Set server42Switch FlowRtt deactivated"
-        def flowRttSwitchFeatureStartState = isEnabledFlowRttForSwitch(server42Switch);
-        disableFlowRttForSwitch(server42Switch)
+        and: "server42FlowRtt for src switch is turned off on src and dst"
```

Comment:
looks like it should be paraphrased
probably to: ` and: "server42FlowRtt is turned off on src and dst"`